### PR TITLE
Support alarm labels

### PR DESCRIPTION
**Pull request by dagmom**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

#5124

### CODE REVIEW

**Comment by wu-sheng** [comment]
And your branch has not been sync from the master, and you don't allow us to sync.

**Comment by dagmom** [comment]
> And your branch has not been sync from the master, and you don't allow us to sync.

@wu-sheng 
Invite you or an org to my fork repository ? or others way

**Comment by wu-sheng** [comment]
> Invite you or an org to my fork repository ? or others way

I am not asking for that, I am just speaking a status. You could update the branch by yourself. The point is, unless. you update the branch with master, we can't merge your PR.

This is what I saw.
image

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java [hunk 1]
```diff
@@ -64,6 +66,8 @@ public String id() {
     private long startTime;
     @Column(columnName = ALARM_MESSAGE, matchQuery = true)
     private String alarmMessage;
+    @Column(columnName = LABELS, matchQuery = true)
+    private String labels;
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java:70]: Adding this? Are you going to modify UI?

Comment (dagmom) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java:70]: I'm not going to modify UI, just store data.
I've just read the code about `@ Column` in other persistent scenario.

```java
@Column(columnName = LABELS, dataType = Column.ValueDataType.LABELED_VALUE)
private DataTable labels;
```

Is that more appropriate? But I have a question, how to search by the specified label(key=value)？

**Comment by dagmom** [comment]
image

**Comment by wu-sheng** [comment]
I know, you set label-values(s) in the entity. But my point is, this can't show up, unless the page changed, right?

**Comment by dagmom** [comment]
> I know, you set label-values(s) in the entity. But my point is, this can't show up, unless the page changed, right?

Yes, do i need to finish UI in this PR? or next PR about UI and query.

**Comment by wu-sheng** [comment]
Please make sure CI passed first, then you need a query protocol PR and UI related PR. Right?

FYI, we are closing the 8.1.0 release, so this PR will be blocked until we are ready to tagging the v8.1.0. So, make your time well planned, you have much time for query protocol and UI PRs.

**Comment by wu-sheng** [comment]
For finishing this PR, you at least need to query-protocol PR merged and implement here. 
Also, considering you have enough time, please add an e2e check for the alarm case. You could take a reference from the e2e TTL, which is using the mesh telemetry to mock data, you could use the same way to mock the data which could trigger the alarm. Then you have everything you need to check. Related to #4530

**Comment by dagmom** [comment]
> For finishing this PR, you at least need to query-protocol PR merged and implement here.
> Also, considering you have enough time, please add an e2e check for the alarm case. You could take a reference from the e2e TTL, which is using the mesh telemetry to mock data, you could use the same way to mock the data which could trigger the alarm. Then you have everything you need to check. Related to #4530

ok

**Comment by wu-sheng** [comment]
No update for a long time.

**Pull request closed without merging**